app/Http/Controllers/Api/LinkController.php

```diff
@@ -26,25 +27,32 @@ public function index()
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function show($id)
+    public function show(Request $request, $id)
     {
-        $link = Link::where('id', $id)->first();
-
-        if(count($link) > 0){
-          return $link;
-        }else{
-          return response('Link Not Found', 404);
-        }
+      $user = $request->user();
+      $link = Link::where('id', $id)->firstOrFail();
+      return $link;
```

Comment:
So adding the request doesn't even come into play here... im assuming you wanted to verify that the user owns the link? but you didn't do that here...